### ISSUE TITLE
Watch asset RPC not accepting symbol with more than 6 characters

### ISSUE THREAD

**Issue by NikosFotiadis** [MetaMask/metamask-extension#10793]
## Bug description
Adding an asset with a symbol larger than 6 characters using the [RPC watch asset](https://docs.metamask.io/guide/rpc-api.html#wallet-watchasset) throws an error but using the extension UI works.

## Steps to reproduce

1. Use the [RPC watch asset](https://docs.metamask.io/guide/rpc-api.html#wallet-watchasset) to add an asset with a symbol with more than 6 characters

## Expected behavior
Should accept the asset since it is also accepted when using the UI

## Browser details (please complete the following information):
 - OS: Ubuntu 20.04
 - Browser: Version 81.0.4044.122 (Official Build) (64-bit)
 - MetaMask Version: 9.2.1

I see that this change has been made on develop branch but was not released with v9.2.1. I think it should have been included so that it follows the behavior of the UI.

**Comment by tmashuang** [comment]
Seems that we aren't properly adhering to the [EIP-747 specification](https://eips.ethereum.org/EIPS/eip-747#specification), which states the max symbol characters is 5.

```
interface WatchAssetParameters {
  type: string; // The asset's interface, e.g. 'ERC20'
  options: {
    address: string; // The hexadecimal Ethereum address of the token contract
    symbol?: string; // A ticker symbol or shorthand, up to 5 alphanumerical characters
    decimals?: number; // The number of asset decimals
    image?: string; // A string url of the token logo
  };
}
```

I consider it a bug, but not for increasing the symbol length. Thoughts @danfinlay?

**Comment by NikosFotiadis** [comment]
> Seems that we aren't properly adhering to the [EIP-747 specification](https://eips.ethereum.org/EIPS/eip-747#specification), which states the max symbol characters is 5.
> 
> ```
> interface WatchAssetParameters {
>   type: string; // The asset's interface, e.g. 'ERC20'
>   options: {
>     address: string; // The hexadecimal Ethereum address of the token contract
>     symbol?: string; // A ticker symbol or shorthand, up to 5 alphanumerical characters
>     decimals?: number; // The number of asset decimals
>     image?: string; // A string url of the token logo
>   };
> }
> ```
> 
> I consider it a bug, but not for increasing the symbol length. Thoughts @danfinlay?

I saw EIP-747 but the length was not mentioned anywhere except for the comment so I assumed it would be a problem to have more characters.

Also, I dont see the point of restricting this to 5 characters when an asset can have more, but that's a different discussion regarding EIP-747.

It seems to me, more logical to support the same thing that is supported by metamask UI, which at the moment of writing this is 11 characters

**Comment by tmashuang** [comment]
After looking at the code, it seems to have change recently in #10670 which will most likely be included after v9.3.0, mid-April. Does that change to 11 character token symbol change sufficient enough to close this?

**Comment by NikosFotiadis** [comment]
Yes, just wanted to verify that it will be included in the next version. Thanks
